aws.go

```diff
@@ -83,7 +89,27 @@ func updateConfigWithAmazonSettings(c *gwconfig.Config) error {
 	c.InstanceType = iid.InstanceType
 	c.AvailabilityZone = iid.AvailabilityZone
 
-	return providerSpecificUpdates(c, awsMetadata)
+	err = providerSpecificUpdates(c, awsMetadata)
+	if err != nil {
+		return err
+	}
+
+	// Don't override WorkerLocation if AWS Provider workerpool / AWS
+	// Provisioner worker type definition specify an explicit value.
```

Comment:
Maybe "if configuration specifies an explicit value"?  Since soon worker-runner will supply this, rather than the worker pool definition.